esmvalcore/preprocessor/_time.py

```diff
@@ -432,33 +432,63 @@ def anomalies(cube, period):
         Available periods: 'full', 'season', 'seasonal', 'monthly', 'month',
         'mon', 'daily', 'day'
 
+    standardize: bool, optional
+        If True standardized anomalies are calculated
+
+
     Returns
     -------
     iris.cube.Cube
-        Monthly statistics cube
+        Anomalies cube
     """
     reference = climate_statistics(cube, period=period)
     if period in ['full']:
-        return cube - reference
+        cube = cube - reference
+        if standardize:
+            cube_stddev = climate_statistics(
+                cube, operator='std_dev', period=period)
+            return cube / cube_stddev
+        else:
+            return cube
 
     cube_coord = _get_period_coord(cube, period)
     ref_coord = _get_period_coord(reference, period)
 
     data = cube.core_data()
-    cube_time = cube.coord('time')
     ref = {}
     for ref_slice in reference.slices_over(ref_coord):
         ref[ref_slice.coord(ref_coord).points[0]] = da.ravel(
             ref_slice.core_data())
     cube_coord_dim = cube.coord_dims(cube_coord)[0]
-    for i in range(cube_time.shape[0]):
-        time = cube_time.points[i]
-        indexes = cube_time.points == time
+    for i in range(cube.coord('time').shape[0]):
+        indexes = cube.coord('time').points == cube.coord('time').points[i]
         indexes = iris.util.broadcast_to_shape(indexes, data.shape,
                                                (cube_coord_dim, ))
         data[indexes] = data[indexes] - ref[cube_coord.points[i]]
 
     cube = cube.copy(data)
+
+    # standardize the results if requested
+    if standardize:
+        cube_stddev = climate_statistics(cube,
+                                         operator='std_dev',
+                                         period=period)
+        if not cube.ndim == cube_stddev.ndim:
```

Comment:
Does this ever happen?

Reply:
I hope not, I just wanted to make sure. But maybe it is better to do these checks in the unit tests if needed? I will take it out here.